src/entity/Chest.c

```diff
@@ -103,46 +103,45 @@ void entity_Chest_begin_opening(Entity* entity) {
 
 INCLUDE_ASM(s32, "entity/Chest", entity_Chest_open);
 
-// regalloc after sin_rad
-#ifdef NON_MATCHING
 void entity_Chest_close(Entity* entity) {
     Chest* data = (Chest*)entity->dataBuf;
+    f32 delta;
+
     switch (data->unk_04) {
-	    case 0:
-	        data->unk_0C = 0.0f;
-	        data->unk_04++;
-	        // fallthrough
-	    case 1:
-	        data->unk_0C += 5.0f;
-	        if (data->unk_0C >= 180.0f) {
-	            data->unk_0C = 180.0f;
-	            data->unk_04++;
-	        }
-	        data->unk_08 += 2.6f * sin_rad(data->unk_0C * TAU / 360.0f);
-	        break;
-	    case 2:
-	        data->unk_0C += 1.0f;
-	        if (data->unk_0C >= 185.0f) {
-	            data->unk_0C = 185.0f;
-	            data->unk_06 = 10;
-	            data->unk_04++;
-	        }
-	        data->unk_08 += 2.0f * sin_rad(data->unk_0C * TAU / 360.0f);
-	        break;
-	    case 3:
-	        data->unk_06--;
-	        if (data->unk_06 == 0) {
-	            data->unk_04++;
-	            entity->flags |= 0x4000;
-	        }
-	        break;
-	    case 4: // needed to make gcc create a jumptable
-	        break;
+        case 0:
+            data->unk_0C = 0.0f;
+            data->unk_04++;
+            // fallthrough
+        case 1:
+            data->unk_0C += 5.0f;
+            if (data->unk_0C >= 180.0f) {
+                data->unk_0C = 180.0f;
+                data->unk_04++;
+            }
+            delta = 2.6f * sin_rad(data->unk_0C * TAU / 360.0f);
+            data->unk_08 += delta;
+            break;
+        case 2:
+            data->unk_0C += 1.0f;
+            if (data->unk_0C >= 185.0f) {
+                data->unk_0C = 185.0f;
+                data->unk_06 = 10;
+                data->unk_04++;
+            }
+            delta = 2.0f * sin_rad(data->unk_0C * TAU / 360.0f);
+            data->unk_08 += delta;
+            break;
+        case 3:
+            data->unk_06--;
+            if (data->unk_06 == 0) {
+                data->unk_04++;
+                entity->flags |= 0x4000;
+            }
+            break;
+        case 4: // needed to make gcc create a jumptable
```

Comment:
LGTM except we don't need this comment - we have to do other annoying manual cases elsewhere in the codebase so it's not worth pointing them out